### PR TITLE
Split type opcode validation into separate pass

### PR DESCRIPTION
**Pull request by alan-baker**
Split type opcode validation into a new file. Unified it with the type unique pass.

Reordered opcode checking to match SPIR-V specification sections.

### CODE REVIEW

File: test/val/val_id_test.cpp [hunk 1]
```diff
@@ -1037,23 +1037,6 @@ TEST_F(ValidateIdWithMessage,
                         "component count does not match Result Type <id> '4's "
                         "vector component count."));
 }
-TEST_F(ValidateIdWithMessage, OpConstantCompositeMatrixColumnTypeBad) {
```
Comment (alan-baker) [inline test/val/val_id_test.cpp]: This test gets caught by type checks now.

Comment (dj2) [reply test/val/val_id_test.cpp]: Should we convert it to check for the type error instead of removing it?

Comment (alan-baker) [reply test/val/val_id_test.cpp]: We do have tests separately for that so I figure it was redundant to add again. I can if you think there's value.

**Comment by alan-baker** [comment]
Rebased, squashed and committed as f2a990022a4b99d11c5fd0567a7c16f67d4e3b59

**Pull request closed without merging**